src/npc.cpp

```diff
@@ -294,6 +302,19 @@ void Npc::onCreatureMove(Creature* creature, const Tile* newTile, const Position
 		if (m_npcEventHandler) {
 			m_npcEventHandler->onCreatureMove(creature, oldPos, newPos);
 		}
+
+		if (creature != this) {
+			Player* player = creature->getPlayer();
+
+			// if player is now in range, add to spectators list, otherwise erase
+			if (player->canSee(_position)) {
```

Comment:
`creature` is either `this` or a player pointer != nullptr. I only use the return of `getPlayer()` when `creature` is not `this`, so it is a player pointer. I can't see where it will ever be nullptr.